flink/v1.18/build.gradle

```diff
@@ -255,6 +260,11 @@ project(":iceberg-flink:iceberg-flink-runtime-${flinkMajorVersion}") {
     description = "Test Flink Runtime Jar against Flink ${flinkMajorVersion}"
     group = "verification"
     jvmArgs += project.property('extraJvmArgs')
+
+    // This flag is set to keep the backward compatibility between JUnit 5.10 and 5.11.
+    // Remove this after all Spark tests are migrated to JUnit 5.11.
+    // See https://github.com/apache/iceberg/pull/11262#issuecomment-2395411961
+    jvmArgs += "-Djunit.platform.reflection.search.useLegacySemantics=true"
```

Comment:
at minimum, we should have an issue about this

Reply:
Yes, it's necessary. I'll raise an issue for this.

Reply:
Created and added the issue to this PR.